control_system_stack/setyaw/src/main.py

```diff
@@ -0,0 +1,185 @@
+from kraken_msgs.msg import absoluteRPY
+
+from kraken_msgs.msg import setYawAction as actionMessageSetYawAction
+from kraken_msgs.msg import setYawFeedback
+from kraken_msgs.msg import setYawResult
+from kraken_msgs.msg import setYawGoal
+
+from all_controllers.fuzzy import Fuzzy
+from all_controllers import fuzzyParams as Fparam
+
+# Global variables
+base_yaw = 0.0
+FIRST_ITERATION = True
+prevError = 0.0
+Current_yaw = -1
+Client_goal = None
+
+class SetYaw(object):
+    """ Creates messages that are used to publish feedback/result to client
+        feedback messages :
+            1. Current value of yaw
+            2. Desired_yaw
+            3. Error
+        result   : elapsed_time
+    """
+    _feedback = setYawFeedback()
+    _result = setYawResult()
+
+    def __init__(self, name):
+        self._action_name = name
+        self._as = actionlib.SimpleActionServer(
+            self._action_name, actionMessageSetYawAction, execute_cb=self.execute_cb, auto_start=False)
+        self._as.start()
+        rospy.loginfo('Server has started. Waiting for a goal.')
+
+
+    def imuCB(self, dataIn):
+        """
+        This is imu call back function.
+
+        1. Updates the Current value of yaw - Current_yaw
+        2. Calculates error and delta_error based on 4 quadrant tangent = arctan2()
+        3. Updates feedback to be sent to the client
+            - Desired_yaw
+            - Current_yaw
+            - Error
+        4. Debug messages
+        """
+        global Current_yaw
+        global base_yaw
+        global prevError
+        global Client_goal
+        global FIRST_ITERATION
+
+        Current_yaw = dataIn.yaw
+
+        if  FIRST_ITERATION :
+    		base_yaw = Current_yaw
+    		FIRST_ITERATION = False
+
+        prevError = YAW.error
+        error = (base_yaw + Client_goal.yaw - Current_yaw)* 3.14 / 180
+        YAW.error = np.arctan2(sin(error),cos(error))*180/3.14
+        YAW.delta_error = YAW.error - prevError
+
+        # Update the feedbacks to be sent
+        self._feedback.Desired_yaw = (Client_goal.yaw + base_yaw)%360
+        self._feedback.Current_yaw = Current_yaw
+        self._feedback.Error = YAW.error
+
+        # calculate the thrust from fuzzy control
+        thrust = YAW.run()
+
+        thruster6Data.data[0] = 0.0
+        thruster6Data.data[1] = 0.0
+        thruster6Data.data[2] = 0.0
+        thruster6Data.data[3] = 0.0
+        thruster6Data.data[4] =  thrust       # Left Thruster
+        thruster6Data.data[5] = -1 * thrust   # Rigt Thruster
+
+        thruster4Data.data[0] = thruster6Data.data[0]
+        thruster4Data.data[1] = thruster6Data.data[1]
+        thruster4Data.data[2] = thruster6Data.data[4]
+        thruster4Data.data[3] = thruster6Data.data[5]
+
+        # publish thrust values to thruster converter
+        pub_thrusters4.publish(thruster4Data)
+        pub_thrusters6.publish(thruster6Data)
+
+        # Debug messages
+        # rospy.logdebug("--------")
```

Comment:
@aniket11bh You don't need to comment out `logdebug` lines. Uncomment these lines, so that if the node is run in `verbose` mode then we will be able to see these lines as well. Otherwise these lines won't show up.